### PR TITLE
add extractLicenseFromLicenseUrl util

### PR DESCRIPTION
**Pull request by dabutvin**
- hook up to the NuGet ClearlyDefined summarizer

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -67,17 +67,59 @@ function addArrayToSet(array, set, valueExtractor) {
...
+  },
+  {
+    test: /^\w*http(s)?:\/\/(www.)?apache.org\/licenses\/LICENSE-2\.0(.html)?$/i,
+    license: 'Apache-2.0'
+  },
+  {
+    test: /^\w*https?:\/\/(?:www.)?gnu.org\/licenses\/(.*?)(\.html)?$/i,
+    licenseMatchGroup: 1
+  },
+  {
```
Comment (iamwillbar) [inline lib/utils.js]: This one (and the one below) are Microsoft specific so we shouldn't include them in ClearlyDefined.

File: lib/utils.js [hunk 2]
```diff
@@ -67,17 +67,59 @@ function addArrayToSet(array, set, valueExtractor) {
...
 }
 
+const _licenseUrlOverrides = [
+  {
+    test: /^\w*https?:\/\/(?:www.)?opensource.org\/licenses\/mit-license(\.php)?$/i,
+    license: 'MIT'
+  },
+  {
+    test: /^\w*https?:\/\/(?:www.)?opensource.org\/licenses\/(.*?)(\.php)?$/i,
+    licenseMatchGroup: 1
```
Comment (iamwillbar) [inline lib/utils.js]: Not sure if it gets handled elsewhere but be careful that the URL isn't case-sensitive but our use of SPDX identifiers *may* be. Same problem exists for the GNU URLs below.

Comment (dabutvin) [reply lib/utils.js]: ahhh good point. I'll add a test case for this

Comment (dabutvin) [reply lib/utils.js]: @iamwillbar I brought in the [spdx-license-list](https://www.npmjs.com/package/spdx-license-list) package to validate against to get the right casing all the way through

This is the same package used in the [chive](https://github.com/amzn/tiny-attribution-generator/blob/master/package.json#L20) library

**Pull request merged**